### ISSUE TITLE
New youtube url structure for sharing support

### ISSUE THREAD

**Issue by BenMcGarry** [honestbleeps/Reddit-Enhancement-Suite#3479]
https://www.youtube.com/shared?ci=bRVS5LR1Gi4

Appears there's a new URL structure for sharing videos under some circumstances...

**Comment by thesbros** [comment]
It seems that the id in this URL is different from the actual video id, so we would need to get the video id somehow.

**Comment by Anichale** [comment]
Looks like the API documentation around the new structure for shared videos is pretty sparse. What youtube-dl is doing right now, is downloading the entire webpage then parsing metadata for the video id. Not sure if that's the method RES wants to do it, or wait until maybe a better way is revealed by the API.

**Comment by erikdesjardins** [comment]
Yeah that's a little more hacky than I'd like to go with (and we'd probably need to request permissions, since I doubt the main page sends CORS headers), so I think we should wait until there's a better way.
